File: java/src/main/java/ai/zentinelle/model/RegisterResult.java
package ai.zentinelle.model;

import java.util.List;
import java.util.Map;

/**
 * Result of agent registration.
 */
public class RegisterResult {

    private final String agentId;
    private final String apiKey;
    private final Map<String, Object> config;
    private final List<PolicyConfig> policies;

    private RegisterResult(Builder builder) {
        this.agentId = builder.agentId;
        this.apiKey = builder.apiKey;
        this.config = builder.config != null ? builder.config : Map.of();
        this.policies = builder.policies != null ? builder.policies : List.of();
    }

    public String getAgentId() { return agentId; }
    public String getApiKey() { return apiKey; }
    public Map<String, Object> getConfig() { return config; }
    public List<PolicyConfig> getPolicies() { return policies; }

    public static Builder builder() { return new Builder(); }

    public static class Builder {
        private String agentId;
        private String apiKey;
        private Map<String, Object> config;
        private List<PolicyConfig> policies;

        public Builder agentId(String agentId) { this.agentId = agentId; return this; }
        public Builder apiKey(String apiKey) { this.apiKey = apiKey; return this; }
        public Builder config(Map<String, Object> config) { this.config = config; return this; }
        public Builder policies(List<PolicyConfig> policies) { this.policies = policies; return this; }

        public RegisterResult build() { return new RegisterResult(this); }
    }
}

File: csharp/src/Zentinelle/Models/RegisterResult.cs
using System.Text.Json.Serialization;

namespace Zentinelle.Models;

/// <summary>
/// Result of agent registration.
/// </summary>
public class RegisterResult
{
    /// <summary>
    /// The registered agent ID.
    /// </summary>
    [JsonPropertyName("agent_id")]
    public string AgentId { get; set; } = string.Empty;

    /// <summary>
    /// The runtime API key returned after bootstrap registration.
    /// </summary>
    [JsonPropertyName("api_key")]
    public string? ApiKey { get; set; }

    /// <summary>
    /// Runtime configuration for the agent.
    /// </summary>
    [JsonPropertyName("config")]
    public Dictionary<string, object> Config { get; set; } = new();

    /// <summary>
    /// Effective policies returned during registration.
    /// </summary>
    [JsonPropertyName("policies")]
    public List<PolicyConfig> Policies { get; set; } = new();
}

/// <summary>
/// Options for agent registration.
/// </summary>
public class RegisterOptions
{
    /// <summary>
    /// Declared agent capabilities.
    /// </summary>
    public List<string> Capabilities { get; set; } = new();

    /// <summary>
    /// Additional metadata for the agent.
    /// </summary>
    public Dictionary<string, object> Metadata { get; set; } = new();

    /// <summary>
    /// Optional display name for the agent.
    /// </summary>
    public string? Name { get; set; }
}
